src/Workspaces/Remote/ServiceHub/Services/CodeAnalysisService_AddImport.cs

```diff
@@ -49,40 +49,40 @@ internal partial class CodeAnalysisService : IRemoteAddImportFeatureService
         /// 
         /// Ideally we would not need to bounce back to the host for this.
         /// </summary>
-        private class SymbolSearchService : ISymbolSearchService
+        private sealed class SymbolSearchService : ISymbolSearchService
         {
-            private readonly CodeAnalysisService codeAnalysisService;
+            private readonly RemoteEndPoint _endPoint;
 
-            public SymbolSearchService(CodeAnalysisService codeAnalysisService)
+            public SymbolSearchService(RemoteEndPoint endPoint)
             {
-                this.codeAnalysisService = codeAnalysisService;
+                _endPoint = endPoint;
             }
 
             public async Task<IList<PackageWithTypeResult>> FindPackagesWithTypeAsync(
                 string source, string name, int arity, CancellationToken cancellationToken)
             {
-                var result = await codeAnalysisService.InvokeAsync<IList<PackageWithTypeResult>>(
-                    nameof(FindPackagesWithTypeAsync), new object[] { source, name, arity }, cancellationToken).ConfigureAwait(false);
-
-                return result;
+                return await _endPoint.InvokeAsync<IList<PackageWithTypeResult>>(
+                    nameof(AbstractAddImportFeatureService<SyntaxNode>.RemoteSymbolSearchService.FindPackagesWithTypeAsync),
```

Comment:
can we not use nameof(IRemoteSymbolSearchUpdateEngine.FindPackagesWithTypeAsync).  That was someone of the point of these interfaces.

Reply:
Yes, we can. I missed that. This call chain is complicated...